pypeit/core/skysub.py

```diff
@@ -675,29 +675,75 @@ def local_skysub_extract(sciimg, sciivar, tilts, waveimg, global_sky, rn2_img,
     # TODO Can this be simply replaced with spat_img above (but not spat_pix since that could have holes)
     spatial_img = thismask * ximg * (np.outer(xsize, np.ones(nspat)))
 
```

Comment:
Can you add some more comments on what the philosophy behind the graph search and depth-first search, along with what this code is trying to achieve. The previous code was admittedly lacking such documentation, but as many (including yours truly) are not that familiar with graphs.

Reply:
I put some comments in `SpecObjs.get_extraction_groups` and `utils.DFS`, can you take a look and let me know if that looks sufficient.